share/tools/web_config/sample_prompts/sorin.fish

```diff
@@ -1,61 +1,219 @@
+    if test -n "$SSH_TTY"
+        echo -n (set_color brred)"$USER"(set_color white)'@'(set_color yellow)(prompt_hostname)' '
+    end
 
-    # Main
-    echo -n (set_color cyan)(prompt_pwd) (set_color red)'❯'(set_color yellow)'❯'(set_color green)'❯ '
+    echo -n (set_color blue)(prompt_pwd)' '
+
+    set_color -o
+    if test "$USER" = 'root'
+        echo -n (set_color red)'# '
+    end
+    echo -n (set_color red)'❯'(set_color yellow)'❯'(set_color green)'❯ '
+    set_color normal
 end
 
 function fish_right_prompt
-    # last status
-    test $status != 0
-    and printf (set_color red)"⏎ "
-
-    if set -l git_dir (git rev-parse --git-dir 2>/dev/null)
-        # Magenta if branch detached else green
-        set -l branch (command git branch -qv | string match "\**")
-        string match -rq detached -- $branch
-        and set_color brmagenta
-        or set_color brgreen
-
-        git name-rev --name-only HEAD
-
-        # Merging state
-        test -f "$git_dir/MERGE_HEAD"
-        and printf ':'(set_color red)'merge'
-        printf ' '
-
-        # Symbols
-        if set -l count (command git rev-list --count --left-right $upstream...HEAD 2>/dev/null)
-            echo $count | read -l ahead behind
-            if test "$ahead" -gt 0
-                printf (set_color magenta)⬆' '
-            end
-            if test "$behind" -gt 0
-                printf (set_color magenta)⬇' '
-            end
+    set -l cmd_status $status
+    if test $cmd_status -ne 0
+        echo -n (set_color red)"✘ $cmd_status "
+    end
+
+    # Get the git directory for later use.
+    # Return if not inside a Git repository work tree.
+    set -l git_dir (command git rev-parse --git-dir ^&-)
+    if test $status -ne 0
+        set_color normal
+        return
+    end
+
+    # Return if zprezto's git-info is disabled.
+    set -l git_info_enabled (command git config --bool prompt.showinfo)
+    if test "$git_info_enabled" = 'false'
+        set_color normal
+        return
+    end
+
+    set -l commit ''
+    if set -l action (_git_action "$git_dir")
+        set commit (command git rev-parse --short=7 HEAD 2> /dev/null)
+    end
+
+    # Get either the branch name or a branch descriptor.
+    set -l branch_detached 0
+    if not set -l branch (command git symbolic-ref --short HEAD ^&-)
+        set branch_detached 1
+        set branch (command git describe --contains --all HEAD ^&-)
+    end
+
+    # Get the commit difference counts between local and remote.
+    set -l IFS \t
+    command git rev-list --count --left-right 'HEAD...@{upstream}' ^&- \
+        | read -l status_ahead status_behind
+    if test $status -ne 0
+        set status_ahead 0
+        set status_behind 0
+    end
+
+    # Get the stash status.
+    # (git stash list) is very slow. => Avoid using it.
+    set -l status_stashed 0
+    if test -f "$git_dir/refs/stash"
+        set status_stashed 1
+    else if test -r "$git_dir/commondir"
+        set -l IFS \n
```

Comment:
If you don't do the IFS setting before, this can be removed.

Reply:
What if the user sets the IFS variable? Wouldn't that affect the theme as well?

Reply:
If the user sets IFS globally, then yes, it will break.

But then everything will break and the user gets to keep the pieces.

This isn't POSIX sh, our IFS isn't really the same thing (even though it looks like it). Since we don't do word-splitting, we really only use $IFS in two places:

- `read`, which now has a "-d"/"--delimiter=" option that takes precedence.
- Command substitutions, where we don't use the _actual value_, but only if the variable is set (in which case we split on newlines) or not (in which case we don't split). This can be better done via `string split0`, which will cause the command substitution to split on NULs (which can't be passed as arguments anyway).

We want to move away from $IFS - _because_ it can be set globally - which is why we've introduced these alternatives in fish 3.

Reply:
> But then everything will break and the user gets to keep the pieces.
I personally can't agree with that. IMO, since a theme is just that, it should be robust against all (or almost all) commands a user enters without breaking.

As such... Would you mind if I keep this as it is?
It already uses the `-d \n` flag you suggested and it shouldn't make a difference to fish if the flag is there or not, right? :slightly_smiling_face: 
Basically I'd just like to be explicit here: I want to read an entire line (potentially the whole file) and not just a whitespace-separated word.

If you have another idea how to write this code more elegantly I'm all for it though. :slightly_smiling_face: